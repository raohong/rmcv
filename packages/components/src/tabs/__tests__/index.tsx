import { render, screen } from '@testing-library/react';
import { Tabs, tabsClassNames } from '..';

const testId = 'tabs';

it('render correctly', () => {
  const tree = render(<Tabs items={[{ tab: 1 }]}></Tabs>);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with defaultActiveKey', () => {
  render(
    <Tabs
      data-testid={testId}
      items={[
        {
          tab: '1',
        },
        {
          tab: '2',
        },
        {
          tab: '3',
        },
      ]}
      defaultActiveKey='3'
    />,
  );

  const tabNav = screen.getAllByRole('tabpanel');

  tabNav.forEach((item, index) => {
    if (index === 2) {
      expect(item).toHaveStyle({
        display: 'none',
      });
    }
    else {
      expect(item).not.toHaveStyle({ display: 'none' });
    }
  });
});

it('render with activeKey', () => {
  render(
    <Tabs
      data-testid={testId}
      items={[
        {
          tab: '1',
        },
        {
          tab: '2',
        },
      ]}
      activeKey='2'
    />,
  );

  const target = screen.getAllByRole('tab')[1];

  expect(target).toHaveAttribute('tabindex', '0');
});

it('render with titleActiveColor and titleInactiveColor', () => {
  render(
    <Tabs
      data-testid={testId}
      titleActiveColor='red'
      items={[
        {
          tab: '1',
        },
        {
          tab: '2',
        },
      ]}
      titleInactiveColor='blue'
    />,
  );

  const [t1, t2] = screen.getAllByRole('tab');

  expect(t1).toHaveStyleRule('color', 'red');
  expect(t2).toHaveStyleRule('color', 'blue');
});

it('render with color and backgroundColor', () => {
  const app = render(
    <Tabs
      color='red'
      backgroundColor='blue'
      items={[
        {
          tab: '1',
        },
        {
          tab: '2',
        },
      ]}
    />,
  );

  const [t1] = screen.getAllByRole('tab');

  expect(t1).not.toHaveStyleRule('background', 'blue');

  app.rerender(
    <Tabs
      color='red'
      backgroundColor='blue'
      items={[
        {
          tab: '1',
        },
        {
          tab: '2',
        },
      ]}
      type='card'
    />,
  );
  const [tab1, tab2] = screen.getAllByRole('tab');

  expect(tab1).toHaveStyleRule('background', 'blue');
  expect(tab2).toHaveStyleRule('color', 'red');
});

it('render with lazyRender', () => {
  render(
    <Tabs
      data-testid={testId}
      items={[
        {
          tab: '1',
        },
        {
          tab: '2',
        },
      ]}
      lazyRender
    />,
  );

  const target = screen
    .getByTestId(testId)
    .querySelectorAll(`.${tabsClassNames.panel}`)[1];

  expect(target).toBeEmptyDOMElement();
});
