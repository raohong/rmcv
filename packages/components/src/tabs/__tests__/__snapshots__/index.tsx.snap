// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  <div
    class="rmcv-tabs rmcv-tabs-line"
  >
    <div
      class="rmcv-tabs-nav"
      role="tablist"
    >
      <div
        class="rmcv-tabs-nav-list"
      >
        <button
          aria-controls="rmcv-tabs-1-tabpanel-0"
          aria-selected="true"
          class="rmcv-tabs-tab rmcv-tabs-tab-active"
          id="rmcv-tabs-1-tab-0"
          role="tab"
          tabindex="0"
        >
          <div
            class="rmcv-tabs-tab-content rmcv-tabs-tab-ellipsis"
          >
            1
          </div>
        </button>
        <button
          aria-controls="rmcv-tabs-1-tabpanel-1"
          aria-selected="false"
          class="rmcv-tabs-tab"
          id="rmcv-tabs-1-tab-1"
          role="tab"
          tabindex="-1"
        >
          <div
            class="rmcv-tabs-tab-content rmcv-tabs-tab-ellipsis"
          >
            2
          </div>
        </button>
      </div>
    </div>
    <div
      class="rmcv-tabs-content"
    >
      <div
        aria-labelledby="rmcv-tabs-1-tabpanel-0"
        class="rmcv-tabs-panel"
        id="rmcv-tabs-1-tabpanel-0"
        role="tabpanel"
      />
      <div
        aria-labelledby="rmcv-tabs-1-tabpanel-1"
        class="rmcv-tabs-panel"
        id="rmcv-tabs-1-tabpanel-1"
        role="tabpanel"
        style="display: none;"
      />
    </div>
  </div>
</DocumentFragment>
`;
